lightly/active_learning/scorers/detection.py

```diff
@@ -43,42 +44,72 @@ def _object_frequency(model_output: List[Dict],
     return np.asarray(scores)
 
 
+def _prediction_margin(model_output: List[ObjectDetectionOutput]):
+    """Score which prefers samples with low max(class prob) * objectness.
+
+    Args:
+        model_output:
+            Predictions of the model of length N.
+
+    Returns:
+        Numpy array of length N with the computed scores.
+
+    """
+    scores = []
+    for output in model_output:
+        if len(output.scores) > 0:
+            # prediction margin is 1 - max(class probs), therefore the mean margin
+            # is mean(1 - max(class probs)) which is 1 - mean(max(class probs))
+            score = 1. - np.mean(output.scores)
+        else:
+            # set the score to 0 if there was no bounding box detected
+            score = 0.
+        scores.append(score)
+    return np.asarray(scores)
+
+
 class ScorerObjectDetection(Scorer):
-    """Class to compute active learning scores from the model_output of a object detection task.
+    """Class to compute active learning scores from the model_output of an object detection task.
 
     Attributes:
         model_output:
-            List of length N containing the predictions of an object detection
-            model. For each sample we expect a dictionary containing `boxes`,
-            `object_probabilities` and `class_probabilities`.
-            `boxes` has shape [B, 4] for B found bounding boxes and the 4 values
-            x1, y1, x2, y2.
-            `object_probabilities` has shape [B] with a objectness value for each
-            bounding box.
-            `class_probabilities` has shape [B, C] for the B bounding boxes and
-            C classes. The class probabilities should all sum up to 1.
+            List of model outputs in an object detection setting.
         config:
             A dictionary containing additional parameters for the scorers.
 
     Examples:
-        >>> predictions = [{
-        >>>     'boxes': [[14, 16, 52, 85]],
-        >>>     'object_probabilities': [0.1024],
-        >>>     'class_probabilities': [[0.5, 0.41, 0.09]]
-        >>> }]
-        >>> scorer = ScorerObjectDetection(predictions)
+        >>> # from output with scores
+        >>> detection_outputs = ObjectDetectionOutput(boxes, scores, labels)
```

Comment:
I would expand the example here and start with boxes, scores, and class probabilities or labels. If I just quickly want to work with the scorer I first have to understand the ObjectDetectionOutput. To simplify I would give an example already here. E.g. from model output to scores. So I don't have to read up on the other parts of the framework all the time.